```
bwa mem -M enterococci_pacbio_index /proj/g2019003/nobackup/private/1_Zhang_2017/genomics_data/Illumina/E745-1.L500_SZAXPI015146-56_1_clean.fq.gz \
/proj/g2019003/nobackup/private/1_Zhang_2017/genomics_data/Illumina/E745-1.L500_SZAXPI015146-56_2_clean.fq.gz > pacbio_illumina_aligned.sam \
| samtools sort -o pacbio_illumina_aligned.bam


bwa mem -M -p bwa -t 2 /home/clno2215/canu_outdir.04-04-2019/canu.04-04-2019.contigs.fasta /home/clno2215/stunning-garbanzo/data/genomics/raw_data/01_Illumina_1_clean_fastqc.gz /home/clno2215/stunning-garbanzo/data/genomics/raw_data/02_Illumina_2_clean_fastqc.gz | samtools sort > aln.05-04-2019.bam 



samtools view -bS aln-illumina_to_pac.sam > aln-illumina_to_pac.bam
samtools sort aln-illumina_to_pac.sam > aln-illumina_to_pac.bam
samtools index aln-illumina_to_pac.bam
```

```
# Input sources
input_Canu_Pacbio=/home/miba8458/2020.03_GenomeAnalysisCourse/scratch/2_DNA_canu_PacBio/canu_pacbio.contigs.fasta
source_Ill_files=/home/miba8458/2020.03_GenomeAnalysisCourse/data/raw_ext/link_to_raw_data/genomics_data/Illumina/
in_file1=E745-1.L500_SZAXPI015146-56_1_clean.fq.gz
in_file2=E745-1.L500_SZAXPI015146-56_2_clean.fq.gz

output_folder_name=/home/miba8458/2020.03_GenomeAnalysisCourse/scratch/4_DNA_bwa_onCanu_PacBio
mkdir -p ${output_folder_name}          # creates the output folders if they don't exist yet
####################################

# Code to run
# BWA first needs to construct the FM-index for the contigs resulted from Canu on PacBio sequences
bwa index -p ${output_folder_name}/pacbio_index ${input_Canu_Pacbio}

# Alignment algorithms are invoked with sub-command 'mem' for the BWA-MEM algorithm
# the algorithm works by seeding alignments with maximal exact matches (MEMs) and then extending seeds with the affine-gap Smith-Waterman algorithm (SW).
# -M to flag shorter split hits as secondary to longer hits
# bwa mem pacbio_index ${source_files}${in_file1} ${source_files}${in_file2} > ${output_folder_name}/align_ill_to_pacbio.sam
bwa mem -M ${output_folder_name}/pacbio_index ${source_Ill_files}${in_file1} ${source_Ill_files}${in_file2} > ${output_folder_name}/align_ill_to_pacbio.sam
#samtools sort -o ${output_folder_name}/align_ill_to_pacbio_sorted.bam

# convert 'sam' to 'bam' (-b) and sort the 'bam' file
#samtools view -u ${output_folder_name}/align_ill_to_pacbio.sam -o ${output_folder_name}/align_ill_to_pacbio.bam
# sort the 'bam' file
#samtools sort ${output_folder_name}/align_ill_to_pacbio.sam -o ${output_folder_name}/align_ill_to_pacbio_sorted.bam
# index the sorted bam file
#samtools index ${output_folder_name}/align_ill_to_pacbio_sorted.bam

```

